### PR TITLE
[Form] Compound option explained

### LINKED ISSUE #5213: [Form] Documentation about "compound" is insufficient / missing
Currently, when your search for something like "Symfony form compound" this is the only thing this is showing up: http://symfony.com/doc/current/reference/forms/types/form.html#compound

For a beginner, this explains basically nothing. Would be nice if there was a more detailed documentation or a section in the cookbook.

### PR DESCRIPTION
**Pull request by vudaltsov**
Closes #5213

ping @HeahDude , @wouterj , @xabbuh

### CODE REVIEW

**Review by javiereguiluz — COMMENTED** [review]
I like this change, but I find this description confusing:

```
The compound option determines whether a form can have children,
in other words is a parent form for other forms.
```

I don't think both descriptions are equivalent: *"can have children"* vs *"is the parent of other forms"*. In pseudo-code this would be like `canHaveChildren()` vs `hasChildren()`. Both methods are not equivalent. So, what's the most precise description of `compound form`?

**Comment by wouterj** [comment]
@javiereguiluz I think that's more due to non-nativeness. In the second statement, "is" should be "may" or "can be": In other words, can be a parent form for other forms.

**Comment by vudaltsov** [comment]
Yes, agree, I will fix this. We also discussed a couple of corrections with @HeahDude, I hope to commit them today.

**Comment by javiereguiluz** [comment]
@vudaltsov could you please finish this PR? I'd love to merge it. Thanks!

**Comment by vudaltsov** [comment]
@javiereguiluz , does it look better now?
I removed everything I was trying to tell about data in the previous commit. Because it's actually wrong. Compoundness is not about data, it's about the form structure and mostly influences the view, not data.

ping @wouterj , @xabbuh , @HeahDude

**Review by xabbuh — APPROVED** [review]

File: reference/forms/types/options/compound.rst.inc [hunk 1]
```diff
@@ -3,6 +3,20 @@ compound
...
+Compound forms are structural elements in a form tree.
+Only a compound form can have children, in other words can be a parent for other forms.
+
+In a view a compound form is rendered as a ``<div>`` container or
+a ``<form>`` element (the whole form is obviously a compound form).
+
+Non-compound forms are always leaves in a form tree, they cannot have children.
+
+A non-compound form is rendered as one of the html form elements: ``<input>``
+(``TextType``, ``FileType``, ``HiddenType``), ``<textarea>`` (``TextareType``)
```
Comment (xabbuh) [inline reference/forms/types/options/compound.rst.inc]: typo: `TextareaType`

File: reference/forms/types/options/compound.rst.inc [hunk 2]
```diff
@@ -3,6 +3,20 @@ compound
 
 **type**: ``boolean`` **default**: ``true``
 
-This option specifies if a form is compound. This is independent of whether
-the form actually has children. A form can be compound but not have any
-children at all (e.g. an empty collection form).
+The compound option determines whether a form is compound.
```
[the remark this answers is not in the document]
Comment (vudaltsov) [reply reference/forms/types/options/compound.rst.inc]: Ok, agree with your comments, hope to find time soon!

**Comment by vudaltsov** [comment]
@javiereguiluz , I feel like I am ready to write an article about this option. Ahahah
I realized that it is one of the clues to understanding the philosophy behind the symfony forms.

**Comment by javiereguiluz** [comment]
@vudaltsov thanks for your work on this. After your last changes, even I can fully understand what this option means and why/when should I use it. Thanks!

**Pull request closed without merging**